### PR TITLE
typescript version 2

### PR DESCRIPTION
**Pull request by rachfop**
## What does this PR do?
This adds additional TS .md files

### CODE REVIEW

File: docs/typescript/how-to-strucutre-a-project-in-typescript.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+id: how-to-strucutre-a-project-in-typescript
```
Comment (flossypurse) [inline docs/typescript/how-to-strucutre-a-project-in-typescript.md:2]:
```suggestion
id: how-to-structure-a-project-in-typescript
```

I think the file name itself has a type in it.

File: docs/typescript/how-to-set-an-activity-retry-policy-in-typescript.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+description: Set an Activity Retry Policy
+---
+
+To set Activity Retry Policies in TypeScript, specify the `proxyActivities`, then add your configurable [Retry Policy](https://typescript.temporal.io/api/interfaces/proto.coresdk.common.iretrypolicy).
+
+```typescript
+// Sample of typical options you can set
+const { greet } = proxyActivities<typeof activities>({
+  startToCloseTimeout: '30s', // recommended
+  scheduleToCloseTimeout: '5m', // useful
```
Comment (flossypurse) [inline docs/typescript/how-to-set-an-activity-retry-policy-in-typescript.md:14]:
```suggestion
  // ...
```

File: docs/typescript/how-to-set-an-activity-retry-policy-in-typescript.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+---
+id: how-to-set-an-activity-retry-policy-in-typescript
+title: How to set an Activity Retry Policy in TypeScript
+sidebar_label: Set an Activity Retry Policy
+description: Set an Activity Retry Policy
+---
+
+To set Activity Retry Policies in TypeScript, specify the `proxyActivities`, then add your configurable [Retry Policy](https://typescript.temporal.io/api/interfaces/proto.coresdk.common.iretrypolicy).
```
Comment (flossypurse) [inline docs/typescript/how-to-set-an-activity-retry-policy-in-typescript.md:8]: Maybe more like "set the Retry Policy in an options object and then pass the object to the constructor method on a new instance of `proxyActivities`." ?

File: docs/typescript/how-to-set-an-activity-retry-policy-in-typescript.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+```typescript
+// Sample of typical options you can set
+const { greet } = proxyActivities<typeof activities>({
```
Comment (flossypurse) [inline docs/typescript/how-to-set-an-activity-retry-policy-in-typescript.md:12]:
```suggestion
const { yourActivity } = proxyActivities<typeof activities>({
```

File: docs/typescript/how-to-set-a-cron-job-in-typescript.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  - developer-guide
+  - sdk
+  - typescript
+---
+
+You can set each Workflow to repeat on a schedule with the `cronSchedule` option:
+
+```typescript
+const handle = await client.start(scheduledWorkflow, {
+  taskQueue: 'test',
```
Comment (flossypurse) [inline docs/typescript/how-to-set-a-cron-job-in-typescript.md:16]:
```suggestion
  // ...
```

File: docs/typescript/how-to-strucutre-a-project-in-typescript.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+description: Structure a project
+tags:
+  - developer-guide
+  - sdk
+  - typescript
+---
+
+A typical Temporal project written in TypeScript consists of the following components:
+
+- Workflows
```
Comment (flossypurse) [inline docs/typescript/how-to-strucutre-a-project-in-typescript.md:14]: I think I was expecting to see more of a file structure example:

```
project-root/
  activities/
      activity.ts
  workflows/
      workerflow.ts
  workers/
      worker.ts
  starter-script.ts
````

**Review by flossypurse — APPROVED** [review]
some stuff you will want to address, some stuff to think about.

File: docs/typescript/how-to-set-a-heartbeat-timeout-in-typescript.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+title: How to set a Heartbeat Timeout in Typescript
+sidebar_label: Set a Heartbeat Timeout
+description: Set a Heartbeat Timeout
+tags:
+  - developer-guide
+  - sdk
+  - typescript
+---
+
+To set a Heartbeat Timeout, create an instance of `longRunningActivity` and set the `heartbeatTimeout` to a time that the Activity can take to complete. If it takes longer than that, the Activity fails.
```
Comment (lorensr) [inline docs/typescript/how-to-set-a-heartbeat-timeout-in-typescript.md:12]:
```suggestion
To set a Heartbeat Timeout, use [`ActivityOptions.heartbeatTimeout`](https://typescript.temporal.io/api/interfaces/common.ActivityOptions#heartbeattimeout). If the Activity takes longer than that between heartbeats, the Activity is failed.
```
Is there a language-agnostic definition of Heartbeat Timeout that shows up in the Guide before this snippet?

File: docs/typescript/how-to-strucutre-a-project-in-typescript.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+- Activities
+- Worker that executes Workflows and Activities
+- A script to execute a Workflow (using a Temporal Client)
+
+Activities cannot be in the same file as Workflows and must be separately registered.
+Activities may be retried repeatedly, so you may need to use [idempotency keys](https://stripe.com/blog/idempotency) for critical side effects.
```
Comment (lorensr) [inline docs/typescript/how-to-strucutre-a-project-in-typescript.md:20]:
```suggestion
Activities may be retried, so you may need to use [idempotency keys](https://stripe.com/blog/idempotency) for critical side effects.
```

Comment (lorensr) [reply docs/typescript/how-to-strucutre-a-project-in-typescript.md:20]: seems redundant

File: docs/typescript/how-to-strucutre-a-project-in-typescript.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  - sdk
+  - typescript
+---
+
+A typical Temporal project written in TypeScript consists of the following components:
+
+- Workflows
+- Activities
+- Worker that executes Workflows and Activities
+- A script to execute a Workflow (using a Temporal Client)
```
Comment (lorensr) [inline docs/typescript/how-to-strucutre-a-project-in-typescript.md:17]:
```suggestion
- Code that uses the Client to execute a Workflow
```

**Review by lorensr — APPROVED** [review]

**Comment by lorensr** [comment]
Is `docs/application-development-guide.md` generated? If so, I put some suggested edits in the wrong place

**Pull request merged**